open-in-vscode/src/open-in-vscode.c

```diff
@@ -111,36 +157,18 @@ GList *menuProviderGetSelectedItems(NautilusMenuProvider *provider, GtkWidget *w
   }
 
   char buttonLabel[32];
+  size_t nrOfDirs = 0ul;
   size_t nrOfFiles = 0ul;
 
-  for (GList *element = files; element != NULL; element = element->next) {
-    NautilusFileInfo *file = element->data;
-
-    if (nautilus_file_info_is_mime_type(file, "text/*"))
-    {
-      char *absoluteFilePath = g_file_get_path(nautilus_file_info_get_location(file));
-
-      command = addToCommand(command, absoluteFilePath);
-      assert(command);
-      ++nrOfFiles;
-
-      g_free(absoluteFilePath);
-    }
-  }
+  handleMimeType("inode/*", &command, files, &nrOfDirs);
+  handleMimeType("text/*", &command, files, &nrOfFiles);
 
-  if (nrOfFiles == 0)
+  if (nrOfDirs == 0 && nrOfFiles == 0)
   {
     return NULL;
   }
 
-  if (nrOfFiles == 1)
-  {
-    strcpy(buttonLabel, "Open file in VSCode");
-  }
-  else
-  {
-    sprintf(buttonLabel, "Open %ld files in VSCode", nrOfFiles);
-  }
+  sprintf(buttonLabel, "%s", buildButtonLabel(nrOfDirs, nrOfFiles));
```

Comment:
wouldn't it make more sense to have buttonLabel as an input/output parameter of `buildButtonLabel`?